hibernate-core/src/main/java/org/hibernate/query/sqm/tree/update/SqmUpdateStatement.java

```diff
@@ -192,15 +194,15 @@ public <Y> SqmUpdateStatement<T> set(Path<Y> attribute, Expression<? extends Y>
 
 	@Override @SuppressWarnings({"rawtypes", "unchecked"})
 	public SqmUpdateStatement<T> set(String attributeName, Object value) {
-		final SqmPath sqmPath = getTarget().get(attributeName);
+		final SqmPath sqmPath = getTarget().get( attributeName );
 		final SqmExpression expression;
 		if ( value instanceof SqmExpression ) {
 			expression = (SqmExpression) value;
 		}
 		else {
-			expression = (SqmExpression) nodeBuilder().value( value );
+			final SqmCriteriaNodeBuilder nodeBuilder = (SqmCriteriaNodeBuilder) nodeBuilder();
+			expression = nodeBuilder.value( value, sqmPath );
 		}
-		assertAssignable( null, sqmPath, expression, nodeBuilder() );
```

Comment:
Shouldn't we still do the assignability check, maybe even within the `applyAssignment` method?

Reply:
We already do that for all assignments in the `SqmUpdateStatement#verifyUpdateTypesMatch`, it was just done twice for that one method.